src/main/java/org/folio/service/orders/CompositePoLineValidationService.java

```diff
@@ -76,6 +76,66 @@ private List<Error> validatePoLineFormats(CompositePoLine compPOL) {
     return Collections.emptyList();
   }
 
+  public List<Error> validatePoLineMaterial(CompositePoLine compPOL) {
+    CompositePoLine.OrderFormat orderFormat = compPOL.getOrderFormat();
+
+    return switch (orderFormat) {
+      case P_E_MIX -> checkP_E_Mix(compPOL);
+      case ELECTRONIC_RESOURCE -> checkElectronicResource(compPOL);
+      case PHYSICAL_RESOURCE -> checkPhysicalResource(compPOL);
+      case OTHER -> checkOtherResource(compPOL);
+      default -> Collections.emptyList();
+    };
+  }
+
+  private List<Error> checkP_E_Mix(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if (compPOL.getPhysical() == null || compPOL.getEresource() == null) {
+      errors.add(ErrorCodes.INVALID_PEMIX_POL);
+    } else if (getElectronicCostQuantity(compPOL) == 0 || getPhysicalCostQuantity(compPOL) == 0) {
+      errors.add(ErrorCodes.INVALID_PEMIX_POL);
```

Comment:
there are you adding same error code, if it is true logic you can join statements

Reply:
right, good call, thank you for catching it :)

Reply:
it changed